src/Integration.Vsix.UnitTests/ErrorList/SonarErrorListDataSource_IssueLocationStoreTests.cs

```diff
@@ -264,6 +250,38 @@ public void RefreshOnBufferChanged_FactoriesWithMatchingFilesUpdatedAndErrorList
             CheckSinkNotifiedOfChangeToFactories(sinkMock2, factoryWithMatch1, factoryWithMatch2);
         }
 
+        [TestMethod]
+        public void RefreshImpl_SomeFactoriesWithMatchingFiles_RaisesSingleIssueChangeEventContainingExpectedFiles()
+        {
+            var testSubject = CreateTestSubject();
+
+            var factoryWithMatch1 = CreateFactoryAndSnapshotWithSpecifiedFiles("match1.aaa", "match2.bbb");
+            var factoryWithMatch2 = CreateFactoryAndSnapshotWithSpecifiedFiles("MATCH1.AAA", "match3.ccc"); // tests case-insensitivity
+            var factoryWithMatch3 = CreateFactoryAndSnapshotWithSpecifiedFiles("not a match but still included.txt", "match4.ddd");
+            var factoryWithMatch4 = CreateFactoryAndSnapshotWithSpecifiedFiles("match5.eee");
+            var factoryWithoutMatches1 = CreateFactoryAndSnapshotWithSpecifiedFiles("not a match.txt");
+            var factoryWithoutMatches2 = CreateFactoryAndSnapshotWithSpecifiedFiles("another not a match.txt");
+
+            testSubject.AddFactory(factoryWithMatch1);
+            testSubject.AddFactory(factoryWithoutMatches1);
+            testSubject.AddFactory(factoryWithMatch2);
+            testSubject.AddFactory(factoryWithoutMatches2);
+            testSubject.AddFactory(factoryWithMatch3);
+            testSubject.AddFactory(factoryWithMatch4);
+
+            var issuesChangedEvent = CreateAndRegisterIssuesEventHandler(testSubject);
+
+            // XXX.txt and YYY.ccc won't match any of the snapshot files
+            testSubject.Refresh(new[] { "XXX.txt", "match1.aaa", "match2.bbb", "match3.ccc", "YYY.ccc", "match4.ddd", "match5.eee" });
+
+            // Should be one event with all directyl and indirectly affected files
+            CheckSingleIssuesChangedEvent(issuesChangedEvent, testSubject,
+                    "match1.aaa", "match2.bbb", "match3.ccc", "match4.ddd", "match5.eee",
+                    // this file is included because it is indirectly included i.e. is in a snapshot a file that does match
```

Comment:
I hadn't thought about indirectly included files and it wasn't tested before.
However, it's consistent with what we are doing when raising the event in other places.